examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+    center[0] = 0.5;
+    center[1] = 0.75;
+    if (dim == 3)
+      {
+        center[2] = 0.5;
+      }
+
+    const double outer_radius = 0.15;
+    const double inner_radius = 0.01;
+
+    parallel::distributed::Triangulation<dim> particle_triangulation(
+      MPI_COMM_WORLD);
+
+    GridGenerator::hyper_shell(
+      particle_triangulation, center, inner_radius, outer_radius, 6);
+    particle_triangulation.refine_global(par.particle_insertion_refinement);
+
+    // We generate the necessary bounding boxes for the particles generator
+    // These bounding boxes are required to quickly identify in which
+    // processors and which cell the inserted particle lies.
+    const auto my_bounding_box = GridTools::compute_mesh_predicate_bounding_box(
+      background_triangulation, IteratorFilters::LocallyOwnedCell());
+    const auto global_bounding_boxes =
+      Utilities::MPI::all_gather(MPI_COMM_WORLD, my_bounding_box);
+
+    // The quadrature points particle generator generates particles only
+    // on locally owned active cells. We therefore count how many of those
+    // are present in the triangulation, as this will be required to
+    // initialize the properties.
+    unsigned int n_locally_owned_cells = 0;
+    for (const auto &cell : particle_triangulation.active_cell_iterators())
+      if (cell->is_locally_owned())
+        n_locally_owned_cells++;
+
+    // We generate an empty vector of properties. We will attribute the
+    // properties to the particles once they are generated.
+    std::vector<std::vector<double>> properties(n_locally_owned_cells,
+                                                std::vector<double>(dim + 1,
+                                                                    0.));
+
+    // We generate the particles at the position of a single
+    // points quadrature. Consequently, one particle will be generated
+    // at the centroid of each cell.
+    Particles::Generators::quadrature_points(particle_triangulation,
+                                             QGauss<dim>(1),
+                                             global_bounding_boxes,
+                                             particle_handler,
+                                             mapping,
+                                             properties);
+
+    pcout << "Number of particles inserted: "
+          << particle_handler.n_global_particles() << std::endl;
+  }
+
+  // @sect4{Background DOFs and interpolation}
+
+  // This function sets up the background degree of freedom used for the
+  // velocity interpolation and allocates the field vector where the entire
+  // solution of the velocity field is stored.
+  template <int dim>
+  void ParticleTracking<dim>::setup_background_dofs()
+  {
+    fluid_dh.distribute_dofs(fluid_fe);
+    IndexSet locally_owned_dofs = fluid_dh.locally_owned_dofs();
+    IndexSet locally_relevant_dofs;
+    DoFTools::extract_locally_relevant_dofs(fluid_dh, locally_relevant_dofs);
```

Comment:
didn't we introduce a function where you can get this index set returned from the dof handler in the same way as for the locally owned dofs?

Reply:
Looking at :
https://www.dealii.org/developer/doxygen/deal.II/classDoFHandler.html
I think there are no such functions. I can always be wrong though :)!.

Reply:
No, you're almost certainly right. We might have *wanted* to do that, but didn't. (See #10563 for something I might have had in mind.)